### ISSUE TITLE
AtomTypeError message is misleading

### ISSUE THREAD

**Issue by rwest** [ReactionMechanismGenerator/RMG-Py#1957]
### Bug Description

Trying to create this molecular structure:
```
no3_h3noh_X =  Molecule().from_adjacency_list("""
no3-_h3noh+
1  O u0 p2 c-1 {2,S} {10,H} {11,D}
2  N u0 p0 c+1 {1,S} {3,D} {4,S}
3  O u0 p2 c0 {2,D}
4  O u0 p3 c-1 {2,S} {7,H}
5  N u0 p0 c+1 {6,S} {8,S} {9,S} {7,S}
6  O u0 p2 c0 {5,S} {10,S}
7  H u0 p0 c0 {4,H} {5,S}
8  H u0 p0 c0 {5,S}
9  H u0 p0 c0 {5,S}
10 H u0 p0 c0 {6,S} {1,H}
11 X u0 p0 c0 {1,D}
""")
no3_h3noh_X
```

fails with the message 
```
AtomTypeError: Unable to determine atom type for atom O-, which has 1 single bonds, 1 double bonds to C, 0 double bonds to O, 0 double bonds to S, 0 triple bonds, 0 benzene bonds, 0 lone pairs, and 2 charge.
```

I'm not saying that my adjacency list is correct or valid, but the issue is that the error message is not helpful: it says I have a single bond to a Carbon, and there are no carbon atoms!  It also says there's a charge of 2, and yet the atom is O-.

### Expected Behavior
The error message would help me debug my problem, rather than confuse me further.

**Comment by amarkpayne** [comment]
I thought this got fixed in #1947 ? Maybe it occurs in multiple places. @xiaoruiDong can you take a quick look?

**Comment by rwest** [comment]
Hmmm. As #1947 was only merged 10 days ago there’s a chance I don’t have it on the branch I was using. If that’s the case then I apologize.  I’ll post here either way once I have time to check.

**Comment by xiaoruiDong** [comment]
I checked the problem, and I think PR #1947 only partially fixes the problem (wrong charge number and lone pair number). If I understand correctly, this atom type refers to the first oxygen atom. If that is the case, then `1 single bond` should be correct (the error message does not distinguish different single bond), but `1 double bond to C` is really confusing. This is related to the following line: https://github.com/ReactionMechanismGenerator/RMG-Py/blob/ea3bc3a750598ad706d8ce167a56013037438cc0/rmgpy/molecule/atomtype.py#L712
If you read the context, you will find RMG assumes double bond should either connect to carbon or oxygen or sulfur. This can be problematic if the other end is N (like -C=N- group) or X in this case (O=X). We can either stop distinguishing different double bonds just like what we did for other bond orders or add "double bond to N" and "double bond to X" in the error message (may looks redundant). Willing to hear suggestion from Dr. West.
